src/components/panels/MachinesPanel/MachinesPanel.js

```diff
@@ -86,6 +90,73 @@ export default function MachinesPanel({
     [modelStatusData, machineId, generateMachinesPanelHeader]
   );
 
+  const filteredModelStatusDataByApp = useCallback(
+    (machineId) => {
+      const filteredModelStatusData = cloneDeep(modelStatusData);
+      filteredModelStatusData &&
+        Object.keys(filteredModelStatusData.applications).forEach(
+          (application) => {
+            const units =
+              filteredModelStatusData.applications[application]?.units;
+
+            if (Object.entries(units).length) {
+              Object.values(units).forEach((unit) => {
+                if (
+                  unit.machine !== machineId ||
+                  !Object.entries(units).length
```

Comment:
Can you add a comment in the code about this as it's not clear why this is being done

Reply:
Done